server/routes.go

```diff
@@ -445,3 +446,21 @@ func putVersionsHandler(m KeyManager, principal knox.Principal, parameters map[s
 		return nil, errF(knox.InternalServerErrorCode, err.Error())
 	}
 }
+
+func authorizeRequest(key *knox.Key, principal knox.Principal, access knox.AccessType) bool {
```

Comment:
Add tests for this function:
* accessCallback true => returns true
* accessCallback false => returns based on key ACL, access type

Reply:
thought about this a lot, i think we should make accessCallback a function parameter, and pass it in higher up. it allows me to mock the functionality of accessCallback

Reply:
nvm, SetAccessCallback already does this so we're fine